### DISCUSSION TITLE
Importing a JSON Schema as a named definition

### DISCUSSION THREAD

**Discussion by jpluscplusm** [cue-lang/cue#1722] [Q&A] [answered]
I can't figure out how to `cue import` a specific JSON Schema into a `#Definition` path.

`cue import`'s `-l <foo>` param seems to want a string path value, meaning I can either use `cue import jsonschema: foo.json -l '"AFoo"'`, which results in the non-definition struct `AFoo`.

Or I can use `cue import jsonschema: foo.json -l '"#AFoo"'` which results in the stringified-named struct `"#AFoo"`, which isn't useable as `foo: #AFoo & { ... }` by the imported schema's consumers.

Obviously I can manually edit the result, on a per-import basis. But is there some invocation subtlety that I'm missing that lets me invoke `cue import` and /not/ have to edit the result?

**Comment by mvdan** [comment, marked as answer]
Perhaps I'm missing something obvious here, but it seems like `-l '#Definition'` works - note that I avoided the double quotes, as otherwise you do get a field whose name includes the hash symbol.

```
$ cue version
cue version v0.4.4-0.20220607133028-5b8860ae5306 linux/amd64
$ cat input.json
{
	"foo": "bar",
	"baz": [3]
}
$ cue import -f -l '#Definition:' json: input.json
$ cat input.cue
#Definition: {
	foo: "bar"
	baz: [3]
}
```

I used `json` rather than `jsonschema` for the sake of keeping the example short. I am assuming and hoping that `-l` doesn't change behavior between import languages :)

**Reply by jpluscplusm** [reply]
Thank you @mvdan - the critical thing, which I'll call out for future-me (when I search for this again!) is this:

**When importing into a specific *and static* path (using only single quotes and not double-quotes-inside-single-quotes), include the trailing colon at the end of the path!**

```
$ cue import file.json --path '#TopLevel:SomeStruct:#ADefinition:'
```

... or more simply, to import into a definition at the top level of the file:

```
$ cue import file.json --path '#ADefinition:'
```

The trailing colon is *critical*: it makes a top-level definition "work" (i.e. the cue-import command doesn't error) and it makes a nested path import Do The Right thing (i.e. my first example (`#TopLevel:SomeStruct:#ADefinition:`), if expressed without the trailing colon, silently discards the `#ADefiniton` suffix, and the cue-import command **doesn't** error) 

Thus, @mvdan, your inline-code example works, but your preamble explanation /doesn't/!
